src/main/scala/com/microsoft/hyperspace/index/sources/default/DefaultFileBasedSource.scala

```diff
@@ -203,7 +210,7 @@ class DefaultFileBasedSource(private val spark: SparkSession) extends FileBasedS
    */
   override def partitionBasePath(location: FileIndex): Option[String] = {
     location match {
-      case p: PartitioningAwareFileIndex =>
+      case p: PartitioningAwareFileIndex if p.partitionSpec.partitions.nonEmpty =>
```

Comment:
return `None` in source provider API means the location is not applicable for this source provider.
So if the check is required in this function, we need to change the return type as `Option[Option[String]]`.
Could you fix the return type? Thanks!

Reply:
do you mean changing the definition to 
```scala
def partitionBasePath(location: FileIndex): Option[Option[String]]
```
if that's the case, isn't it easier to change the description of this function to just be: if none, then partitionBasePath is not defined, for any reason whatsoever. Reason could be location is not compatible or location is not hive-partitioned?

Alternately, if we define Option[Option[String]], could you suggest what these scenarios will mean:
Some(Some(path)) => location is supported and basepath is valid
Some(None) => location is supported but basePath is not valid
None => location is not supported

My only concern is this will create difficulty in understanding of the api. I would suggest it's better to keep it simple meaning:
partitioBasePath = 
Some(path) => for valid scenarios
None => for invalid scenarios for any reason. (I am just clubbing the Some(None) and None above.
Please let me know what you think

Reply:
>Alternately, if we define Option[Option[String]], could you suggest what these scenarios will mean:
Some(Some(path)) => location is supported and basepath is valid
Some(None) => location is supported but basePath is not valid
None => location is not supported

I meant this. It's because all of the source provider return None for a API call, the source provider manager throws `No source provider returned valid results.` exception. 
Otherwise, we should check partitonSpec is empty or not before calling API. In this way, if we call the API with non-partitioned location, it will throw the null exception which is easier to know the reason.

Reply:
> My only concern is this will create difficulty in understanding of the api.

The source provider APIs return `Option[T]` so that only provider is picked for a relation: https://github.com/microsoft/hyperspace/blob/8fdf28b0019fc6b443081202e0c440fd8b908b6b/src/main/scala/com/microsoft/hyperspace/index/sources/FileBasedSourceProviderManager.scala#L123-L133

Thus, if `T` needs to be `Option[String]`, the return value for the API should be `Option[Option[String]]` as @sezruby pointed out. Does it make sense?

Reply:
cool. I have pushed another iteration with this suggestion. Please have a look

Reply:
I'm not fond of `Option[Option[String]]`. It is very hard to understand and very rarely used -- it looks like a coding mistake. Other option would be to create a custom created class/type/object (i.e: `LocationWrapper` or something) to model the 3 states that are needed or maybe use the `Either` class.

If it is still decided to go with `Option[Option[String]]` I would suggest to add to the scala doc the meaning defined above: 

```
Some(Some(path)) => location is supported and basepath is valid
Some(None) => location is supported but basePath is not valid
None => location is not supported
```

Reply:
@andrei-ionescu Fair enough. Do you want to update this as a follow up PR once this PR is checked in? Otherwise, @apoorvedave1 can do a follow up. Please let us know.